Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -771,9 +796,8 @@ void SkylanderPortalWindow::RefreshList()
 
 void SkylanderPortalWindow::CreateSkyfile(const QString& path, bool load_after)
 {
-  auto& system = Core::System::GetInstance();
-
-  if (!system.GetSkylanderPortal().CreateSkylander(path.toStdString(), m_sky_id, m_sky_var))
+  IOS::HLE::USB::SkylanderFigure* figure = new IOS::HLE::USB::SkylanderFigure(path.toStdString());
```

Comment:
You are actually leaking memory here if you `return` during a failure.  Additionally, manually `new`'ing and `delete`ing is error prone.  I'd suggest using a [unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr) instead.